### PR TITLE
Enable collisions for remote palyer / ship bodies

### PR DESCRIPTION
**Pull request by Raicuparta**

### CODE REVIEW

File: QSB/TransformSync/PlayerTransformSync.cs [hunk 1]
```diff
@@ -29,9 +29,20 @@ protected override Transform InitLocalTransform()
...
             var body = Instantiate(GetPlayerModel());
-
             GetComponent<AnimationSync>().InitRemote(body);
 
+            var physicsBody = new GameObject("PlayerBodySync");
+
+            var collider = physicsBody.AddComponent<CapsuleCollider>();
+            collider.radius = 1;
+            collider.height = 2;
+            collider.center = Vector3.up * 1;
```
Comment (amazingalek) [inline QSB/TransformSync/PlayerTransformSync.cs]: I recommend adding a cylinder collider to the top half of the capsule, to make it easy to stand on players heads 😁

Comment (Raicuparta) [reply QSB/TransformSync/PlayerTransformSync.cs]: Any reason to keep the capsule? Might as well just use a cylinder for the whole thing, no?

Comment (amazingalek) [reply QSB/TransformSync/PlayerTransformSync.cs]: Cylinder is great for sliding on the ground, not get stuck on things.

Comment (Raicuparta) [reply QSB/TransformSync/PlayerTransformSync.cs]: Ah ok. In this case these are just kinematic colliders for the remote players, so it doesn't matter if they are good at sliding or not, they'll just always follow the remote player around. So I'll just use a cylinder.

File: QSB/TransformSync/PlayerTransformSync.cs [hunk 2]
```diff
@@ -29,9 +29,20 @@ protected override Transform InitLocalTransform()
...
+
+            var rigidBodySync = physicsBody.AddComponent<RigidbodySync>();
+            rigidBodySync.target = body;
+            rigidBodySync.localColliderType = typeof(PlayerBody);
+            rigidBodySync.IgnoreCollision(Locator.GetShipTransform().gameObject);
```
Comment (amazingalek) [inline QSB/TransformSync/PlayerTransformSync.cs]: Create a Init method on RigidbodySync?

File: QSB/TransformSync/RigidbodySync.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+using System.Linq;
+using System.Text;
+using UnityEngine;
+
+namespace QSB.TransformSync
+{
+    class RigidbodySync : MonoBehaviour
+    {
+        public Transform target;
+        public Type localColliderType;
```
Comment (amazingalek) [inline QSB/TransformSync/RigidbodySync.cs]: if you use an Init method as suggested above, these can be private

**Comment by amazingalek** [comment]
This is very cool, amazing work! 😀

File: QSB/TransformSync/PlayerTransformSync.cs (hunk 1 above)
Comment (Raicuparta) [reply QSB/TransformSync/PlayerTransformSync.cs]: @amazingalek Never realized but apparently cylinder colliders aren't a thing in Unity by default. I think sticking with capsule is fine for now, no reason to make it too complex.

**Pull request closed without merging**